### PR TITLE
Renaming of (co)products

### PR DESCRIPTION
**Pull request by mortberg**
This PR addresses https://github.com/UniMath/UniMath/issues/366 by:

- Renaming (co)product to bin(co)product

- Renaming arbitrary_(co)product to (co)product

- Renaming (co)product_functor defined in FunctorsPointwise(Co)Product to (co)product_of_functors (and similarily for binary versions). The less direct version (defined by composing delta, pair and product functors) is now called product_of_functors_alt.

### CODE REVIEW

File: UniMath/CategoryTheory/.package/files [hunk 1]
```diff
@@ -18,13 +18,13 @@ limits/initial.v
 limits/graphs/initial.v
 limits/zero.v
 limits/graphs/zero.v
+limits/bincoproducts.v
 limits/coproducts.v
-limits/arbitrary_coproducts.v
-limits/graphs/coproducts.v
+limits/graphs/bincoproducts.v
```
Comment (benediktahrens) [inline UniMath/CategoryTheory/.package/files]: Would a camelcase `BinCoproducts.v` be easier to read?

File: UniMath/CategoryTheory/limits/FunctorsPointwiseBinCoproduct.v [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+Variable HD : BinCoproducts D.
+Variable hsD : has_homsets D.
+
+Section bincoproduct_of_functors.
+
+Variables F G : functor C D.
+
+Local Notation "c ⊗ d" := (BinCoproductObject _ (HD c d))(at level 45).
+
+Definition bincoproduct_of_functors_ob (c : C) : D := F c ⊗ G c.
```
Comment (benediktahrens) [inline UniMath/CategoryTheory/limits/FunctorsPointwiseBinCoproduct.v]: The spelling/capitalization should be unified. Since it is `BinCoproduct`, it should probably be `BinCoproduct_of_functors`?

**Comment by benediktahrens** [comment]
I have taken a superficial look, and am ready to trust in Travis.

File: UniMath/CategoryTheory/.package/files (hunk 1 above)
Comment (mortberg) [inline UniMath/CategoryTheory/.package/files]: I prefer all lowercase letters in filenames, but I'm not too fanatic about it (it is not like trailing whitespace... :)). However there seems to be no standard for this in UniMath at the moment. Maybe we could decide on a standard for the CategoryTheory library and make another PR where we make the file names more uniform?

File: UniMath/CategoryTheory/limits/FunctorsPointwiseBinCoproduct.v (hunk 2 above)
Comment (mortberg) [inline UniMath/CategoryTheory/limits/FunctorsPointwiseBinCoproduct.v]: Maybe it should also be camelcase? I've noticed we are not at all consequent about this... However making identifier names more standars should probably be done in another PR. I'll only make the change you suggest for now.

File: UniMath/CategoryTheory/.package/files (hunk 1 above)
Comment (mortberg) [inline UniMath/CategoryTheory/.package/files]: I take that back, the standard seems to be to use camelcase, so let's make another PR where we make the CT library filenames camelcase as well.

Comment (benediktahrens) [inline UniMath/CategoryTheory/.package/files]: Let's say that @DanGrayson imposed his standard on us.

Comment (DanGrayson) [inline UniMath/CategoryTheory/.package/files]: :-)

I was surprised to see "the standard" in a comment, as though my proposal had been approved or even acclaimed.

Comment (vladimirias) [inline UniMath/CategoryTheory/.package/files]: What is camelcase? How does one know how to write in it when one need to import bincoproducts.v?

Comment (DanGrayson) [inline UniMath/CategoryTheory/.package/files]: I didn't know, either: https://en.wikipedia.org/wiki/CamelCase
